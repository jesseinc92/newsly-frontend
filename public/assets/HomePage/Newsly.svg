<svg width="329.24375px" height="121.39999999999999px" xmlns="http://www.w3.org/2000/svg" viewBox="85.37812500000001 14.300000000000004 329.24375 121.39999999999999" style="background: rgba(0, 0, 0, 0);" preserveAspectRatio="xMidYMid"><defs><linearGradient id="editing-gradient-gradient" x1="0.19909248842397587" x2="0.8009075115760241" y1="0.10068224497635359" y2="0.8993177550236464" gradientUnits="objectBoundingBox"><stop offset="0" stop-color="#f0f"></stop><stop offset="1" stop-color="#0ff"></stop></linearGradient></defs><g><g transform="translate(136.58999568223953, 104.47000122070312)"><path d="M7.68 0L0.64 0L8.32-42.24L22.08-42.24L32.06-13.18L37.38-42.24L44.48-42.18L36.67 0L22.02 0L12.80-27.97L7.68 0ZM72-10.82L72-10.82L72-10.82Q73.60-9.73 73.60-7.33L73.60-7.33L73.60-7.33Q73.60-4.93 72.38-3.39L72.38-3.39L72.38-3.39Q71.17-1.86 69.18-0.83L69.18-0.83L69.18-0.83Q65.09 1.28 60.67 1.28L60.67 1.28L60.67 1.28Q56.26 1.28 53.66 0.32L53.66 0.32L53.66 0.32Q51.07-0.64 49.34-2.43L49.34-2.43L49.34-2.43Q45.95-5.82 45.95-12.03L45.95-12.03L45.95-12.03Q45.95-21.70 51.20-27.58L51.20-27.58L51.20-27.58Q56.83-33.92 66.62-33.92L66.62-33.92L66.62-33.92Q72.70-33.92 75.71-31.36L75.71-31.36L75.71-31.36Q77.95-29.44 77.95-26.30L77.95-26.30L77.95-26.30Q77.95-15.04 58.50-15.04L58.50-15.04L58.50-15.04Q58.24-13.38 58.24-11.97L58.24-11.97L58.24-11.97Q58.24-9.02 59.55-7.90L59.55-7.90L59.55-7.90Q60.86-6.78 63.30-6.78L63.30-6.78L63.30-6.78Q65.73-6.78 68.32-7.90L68.32-7.90L68.32-7.90Q70.91-9.02 72-10.82ZM58.94-17.79L58.94-17.79L58.94-17.79Q63.49-17.79 66.11-20.61L66.11-20.61L66.11-20.61Q68.74-23.30 68.74-27.58L68.74-27.58L68.74-27.58Q68.74-29.06 68.19-29.86L68.19-29.86L68.19-29.86Q67.65-30.66 66.56-30.66L66.56-30.66L66.56-30.66Q65.47-30.66 64.54-30.24L64.54-30.24L64.54-30.24Q63.62-29.82 62.66-28.42L62.66-28.42L62.66-28.42Q60.29-25.22 58.94-17.79ZM127.87-10.69L127.87-10.69Q123.65-4.86 119.58-1.79L119.58-1.79L119.58-1.79Q115.52 1.28 111.33 1.28L111.33 1.28L111.33 1.28Q107.14 1.28 105.66 0.45L105.66 0.45L105.66-13.63L105.66-13.63Q102.02-3.90 95.49-0.13L95.49-0.13L95.49-0.13Q93.12 1.28 90.88 1.28L90.88 1.28L90.88 1.28Q86.59 1.28 84.74 0.45L84.74 0.45L84.74 0.45Q84.29-14.21 83.90-18.27L83.90-18.27L83.90-18.27Q83.52-22.34 83.20-24.83L83.20-24.83L83.20-24.83Q82.56-29.82 80.38-31.87L80.38-31.87L80.38-31.87Q83.14-33.92 86.98-33.92L86.98-33.92L86.98-33.92Q94.40-33.92 95.42-24.70L95.42-24.70L95.42-24.70Q95.68-22.53 95.68-20.22L95.68-20.22L95.68-20.22Q95.68-16.77 94.91-4.74L94.91-4.74L94.91-4.74Q96.45-5.44 98.08-7.97L98.08-7.97L98.08-7.97Q99.71-10.50 101.06-13.95L101.06-13.95L101.06-13.95Q104.06-21.95 104.06-28.86L104.06-28.86L104.06-28.86Q104.06-29.95 103.78-31.10L103.78-31.10L103.78-31.10Q103.49-32.26 103.17-32.58L103.17-32.58L103.17-32.58Q105.41-33.92 108.83-33.92L108.83-33.92L108.83-33.92Q112.26-33.92 113.89-31.58L113.89-31.58L113.89-31.58Q115.52-29.25 115.97-24.96L115.97-24.96L115.97-24.96Q116.48-19.52 116.48-16.26L116.48-16.26L116.48-16.26Q116.48-11.71 116.03-4.99L116.03-4.99L116.03-4.99Q119.23-7.30 121.86-14.85L121.86-14.85L121.86-14.85Q124.48-22.40 124.48-27.78L124.48-27.78L124.48-27.78Q124.48-30.14 124.10-31.55L124.10-31.55L124.10-31.55Q126.21-33.92 129.86-33.92L129.86-33.92L129.86-33.92Q132.10-33.92 133.73-32.70L133.73-32.70L133.73-32.70Q135.36-31.49 135.36-29.22L135.36-29.22L135.36-29.22Q135.36-26.94 134.69-24.51L134.69-24.51L134.69-24.51Q134.02-22.08 132.93-19.58L132.93-19.58L132.93-19.58Q130.82-14.59 127.87-10.69L127.87-10.69ZM156.22-22.46L156.22-22.46L156.22-22.46Q157.12-24.83 157.12-26.88L157.12-26.88L157.12-26.88Q157.12-30.66 154.05-30.66L154.05-30.66L154.05-30.66Q152.45-30.66 151.14-29.22L151.14-29.22L151.14-29.22Q149.82-27.78 149.82-25.98L149.82-25.98L149.82-25.98Q149.82-24.70 150.85-23.68L150.85-23.68L150.85-23.68Q152.32-22.27 156.10-19.97L156.10-19.97L156.10-19.97Q159.87-17.66 161.31-15.78L161.31-15.78L161.31-15.78Q162.75-13.89 162.75-11.30L162.75-11.30L162.75-11.30Q162.75-8.70 161.50-6.34L161.50-6.34L161.50-6.34Q160.26-3.97 158.02-2.30L158.02-2.30L158.02-2.30Q153.22 1.28 145.54 1.28L145.54 1.28L145.54 1.28Q141.38 1.28 138.18-0.90L138.18-0.90L138.18-0.90Q134.98-3.01 134.98-5.50L134.98-5.50L134.98-5.50Q134.98-8 136.80-9.47L136.80-9.47L136.80-9.47Q138.62-10.94 141.31-10.94L141.31-10.94L141.31-10.94Q144-10.94 145.66-9.92L145.66-9.92L145.66-9.92Q144.83-7.81 144.83-6.40L144.83-6.40L144.83-6.40Q144.83-2.18 148.42-2.18L148.42-2.18L148.42-2.18Q149.95-2.18 150.98-3.14L150.98-3.14L150.98-3.14Q152-4.10 152-5.76L152-5.76L152-5.76Q152-9.02 146.50-12.48L146.50-12.48L146.50-12.48Q142.02-15.42 140.99-16.58L140.99-16.58L140.99-16.58Q139.26-18.62 139.26-21.18L139.26-21.18L139.26-21.18Q139.26-23.74 140.48-26.18L140.48-26.18L140.48-26.18Q141.70-28.61 143.94-30.34L143.94-30.34L143.94-30.34Q148.61-33.92 156.74-33.92L156.74-33.92L156.74-33.92Q160.90-33.92 163.36-32.26L163.36-32.26L163.36-32.26Q165.82-30.59 165.82-27.78L165.82-27.78L165.82-27.78Q165.82-24.96 164.13-23.36L164.13-23.36L164.13-23.36Q162.43-21.76 159.36-21.76L159.36-21.76L159.36-21.76Q157.25-21.76 156.22-22.46ZM183.49-3.52L183.49-3.52L183.49-3.52Q181.50 1.28 175.17 1.28L175.17 1.28L175.17 1.28Q171.97 1.28 169.92-0.96L169.92-0.96L169.92-0.96Q168.26-2.82 168.26-4.93L168.26-4.93L168.26-4.93Q168.26-9.73 170.50-20.22L170.50-20.22L175.17-44.80L188.16-46.08L181.95-13.70L181.95-13.70Q180.86-8.96 180.86-7.30L180.86-7.30L180.86-7.30Q180.86-3.65 183.49-3.52ZM209.28-7.23L209.28-7.23L209.28-7.23Q205.95-3.78 199.87-3.78L199.87-3.78L199.87-3.78Q192.26-3.78 190.66-9.34L190.66-9.34L190.66-9.34Q190.14-11.01 190.14-12.86L190.14-12.86L190.14-12.86Q190.14-14.72 190.53-16.64L190.53-16.64L193.54-32.64L206.53-33.92L203.01-15.30L203.01-15.30Q202.88-14.46 202.69-13.57L202.69-13.57L202.69-13.57Q202.50-12.67 202.50-11.74L202.50-11.74L202.50-11.74Q202.50-10.82 202.82-10.21L202.82-10.21L202.82-10.21Q203.14-9.60 203.58-9.22L203.58-9.22L203.58-9.22Q204.35-8.70 205.73-8.70L205.73-8.70L205.73-8.70Q207.10-8.70 208.29-10.37L208.29-10.37L208.29-10.37Q209.47-12.03 210.05-14.91L210.05-14.91L213.57-32.64L226.18-33.92L221.38-8.77L221.38-8.77Q219.01 3.52 214.14 8.38L214.14 8.38L214.14 8.38Q211.65 10.88 208.42 11.87L208.42 11.87L208.42 11.87Q205.18 12.86 201.02 12.86L201.02 12.86L201.02 12.86Q194.30 12.86 190.46 10.75L190.46 10.75L190.46 10.75Q186.62 8.64 186.62 5.12L186.62 5.12L186.62 5.12Q186.62 2.50 188.61 0.99L188.61 0.99L188.61 0.99Q190.59-0.51 193.66-0.51L193.66-0.51L193.66-0.51Q196.35-0.51 198.40 0.64L198.40 0.64L198.40 0.64Q199.62 1.34 200.19 2.24L200.19 2.24L200.19 2.24Q198.72 3.52 198.72 5.63L198.72 5.63L198.72 5.63Q198.72 8.45 201.28 8.45L201.28 8.45L201.28 8.45Q205.50 8.45 207.94-1.28L207.94-1.28L207.94-1.28Q208.64-4.16 209.28-7.23Z" fill="url(#editing-gradient-gradient)"></path></g></g><style>text {
  font-size: 64px;
  font-family: Arial Black;
  dominant-baseline: central;
  text-anchor: middle;
}</style></svg>