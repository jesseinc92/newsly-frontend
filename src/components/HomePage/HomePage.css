.HomePage h1 {
  font-family: 'Fjalla One', sans-serif;
  font-size: 4rem;
}


.HomePage-container {
  margin-top: 20vh;
}

/* Tablet screen sizing */
@media screen and (min-width: 768px) {

  .HomePage h1 {
    font-size: 5rem;
  }

}

/* Laptop/desktop sizing */
@media screen and (min-width: 1024px) {
  
  .HomePage h1 {
    font-size: clamp(80px, 7.8vw, 100px);
  }

  
}