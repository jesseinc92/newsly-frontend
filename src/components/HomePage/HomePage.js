import React from 'react';
import Button from '../Button/Button';

const Homepage = () => {
  return (
    <main className='HomePage'>
      <h1>Newsly</h1>
      <p>The mindful app for mindful news consumption.</p>
      <div className='Button-group'>
        <Button text='Read' path='/front-page' />
        <Button text='Login' />
      </div>
    </main>
  )
}

export default Homepage;