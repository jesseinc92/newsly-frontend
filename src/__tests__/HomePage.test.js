import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderWithContext from '../helpers/testContext';
import HomePage from '../components/HomePage/HomePage';

it('renders the components', function() {
  renderWithContext(<MemoryRouter><HomePage /></MemoryRouter>)
});

it('matches the snapshot', function() {
  const { asFragment } = renderWithContext(<MemoryRouter><HomePage /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});

it('does not contain login button where there is a saved user', function() {
  renderWithContext(<MemoryRouter><HomePage /></MemoryRouter>);
  const login = screen.queryByText('Login');
  expect(login).not.toBeInTheDocument();
})

it('contains login button when there is no saved user', function() {
  renderWithContext(<MemoryRouter><HomePage /></MemoryRouter>, null);
  const login = screen.queryByText('Login');
  expect(login).toBeInTheDocument();
});