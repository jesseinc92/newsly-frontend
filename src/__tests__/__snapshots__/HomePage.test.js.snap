// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <main
    class="HomePage"
  >
    <div
      class="HomePage-container"
    >
      <h1>
        Newsly
      </h1>
      <p>
        The mindful app for mindful news consumption.
      </p>
      <div
        class="Button-group"
      >
        <button
          class="Button"
        >
          <a
            href="/front-page"
          >
            Read
          </a>
        </button>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
